docs/developers/design/005_genmagstr_differences.md

```diff
@@ -0,0 +1,223 @@
+
+    $\mathbf{S}^{gen}_i(\mathbf{r}) = R(2 \pi \mathbf{k_m} \cdot \mathbf{r})\cdot \mathbf{S}_i$
+
+    where $S_i$ has either a single moment or as many moments
+          as the number of magnetic atoms in the crystallographic
+          cell. In the first case $r$ denotes the atomic
+          positions, while for the second case $r$ denotes the
+          position of the origin of the cell.
+  * `'rotate'` (optionally reads `S`, `phi`, `phid`, `n`, `nExt`)
+          uniform rotation of all magnetic moments with a
+          `phi` angle around the given `n` vector. If
+          `phi=0`, all moments are rotated so, that the first
+          moment is parallel to `n` vector in case of
+          collinear structure or in case of planar structure
+          `n` defines the normal of the plane of the magnetic
+          moments.
+  * `'func'` (reads `func`, `x0`)
+          function that defines the parameters of the single-k
+          magnetic structure: moment vectors, propagation vector
+          and normal vector from arbitrary parameters in the
+          following form:
+    ```
+    [S, k, n] = @(x)func(S0, x)
+    ```
+    where `S` is matrix with dimensions of $[3\times n_{magExt}]$. `k` is
+          the propagation vector in a 3-element row vector. `n` is the
+          normal vector of the spin rotation plane also 3-element row
+          vector. The default value for `func` is `@gm_spherical3d`. For planar
+          magnetic structure use `@gm_planar`. Only `func` and `x`
+          have to be defined for this mode.
+
+    <span style="color:red">**This mode is completely removed**: the reasoning is 
+     that people that want more control should just use the python interface </span> 
+
+ * `'fourier'` (reads `S`, optionally reads `k`, `r0`, `nExt`, `epsilon`)
+          same as `'helical'`, except the `S` option is taken as the
+          Fourier components, thus if it contains real numbers, it will
+          generate a sinusoidally modulated structure instead of
+          a spiral.
+
+`'phi'`
+: Angle of rotation of the magnetic moments in radian. Default
+  value is 0.
+
+`'phid'`
+: Angle of rotation of the magnetic moments in \\deg. Default
+  value is 0.
+
+`'nExt'`
+:  Size of the magnetic supercell in multiples of the
+  crystallographic cell, dimensions are $[1\times 3]$. Default value is
+  stored in `obj`. If `nExt` is a single number, then the size of the
+  extended unit cell is automatically determined from the first
+  magnetic ordering wavevector. E.g. if `nExt = 0.01`, then the number
+  of unit cells is determined so, that in the extended unit cell,
+  the magnetic ordering wave vector is `[0 0 0]`, within the given
+  0.01 rlu tolerance.
+  
+: <span style="color:red">**Details of the implementation are not the same**: 
+  the way that commensurate and incommensurate vectors are represented is different,
+  and the tolerances are not the same. It should work roughly the same way though.
+  </span>

```

Comment:
`<span>` CSS styles not supported in Github markdown - could you change this to either LaTeX or the github notes blockquote, please?